enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,597 @@
+the 3 nodes for ETCD quorum.
+
+### User Stories
+
+- As a solutions architect for a retail organization, I want to deploy OpenShift
+  at n number of store locations at the edge with only 2 regular sized nodes and
+  1 lower cost node to maintain HA and keep compute costs down.
+- As a solutions architect for cloud infrastructures, I want to offer low cost
+  OpenShift deployments on purpose built hardware for a 2 + 1 configuration.
+- As an OpenShift cluster admin I want non-critical applications deployed to my
+  2 + 1 arbiter node cluster to not be scheduled to run on the arbiter node.
+- As an OpenShift cluster admin I want to be able to allow deployments with
+  proper tolerations or explicitly defined node in the `spec` to be able to be
+  scheduled on the arbiter node.
+
+### Goals
+
+- Provide a new arbiter node role type that achieves HA but does not act as a
+  full master node.
+- HA for a 2+1 arbiter node should match the HA guarantees of a 3 Node Cluster
+  deployment.
+- Support installing OpenShift with 2 master nodes and 1 arbiter node.
+- Add a new ControlPlaneTopology and InfrastructureTopology in the
+  Infrastructure API
+- The arbiter node hardware requirements will be lower than regular nodes in
+  both cost and performance. Customers can use devices on the market from OEMs
+  like Dell that supply an all in one unit with 2 compute and 1 lower powered
+  compute for this deployment scenario.
+
+### Non-Goals
+
+The below goals are not intended to be worked on now, but might be expansion
+ideas for future features.
+
+- Running the arbiter node offsite.
+- Running a virtualized arbiter node on the same cluster.
+- Having a single arbiter supporting multiple clusters.
+- Moving from 2 + 1 to a conventional 3 node cluster
+
+## Proposal
+
+The main focus of the enhancement is to support edge deployments of individual
+OpenShift HA clusters at scale, and to do so in a cost effective way. We are
+proposing doing this through the creation of a new node role type called an
+`node-role.kubernetes.io/arbiter` node as a heterogenous quasi control plane
+configuration. The arbiter will run the critical components that help maintain
+an HA cluster, but other platform pods should not be scheduled on the arbiter
+node. The arbiter node will be tainted to make sure that only deployments that
+tolerate that taint are scheduled on the arbiter.
+
+We think creating this new node role would be the best approach for this
+feature. Having a new node role type allows our bootstrap flow to leverage the
+existing MCO and Installer systems to supply specific Ignition files for a
+smaller footprint device. Existing deployments will essentially be blind to this
+node role unless they need to be made aware of it, but critical workloads that
+allow all taints can still run on the node, thus making it easier for smaller
+arbiter nodes to be deployed with out higher memory and cpu requirements.
+
+Components that we are proposing to change:
```

Comment:
HCP to include ban on installing on this topology.  Probably also ban installing on single node.